woocommerce/woocommerce
# Add the `include_types` param to the `/products` endpoint

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR introduces the new `include_types` parameter to the `/products` endpoint.
It allows filtering products by multiple types simultaneously: `simple`, `variable`, `external`, and `grouped`.

With this implementation, `include_types` takes precedence over the `type` param when both are present.

Closes #52944



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure you have products of all types (`simple`, `variable`, `external`, and `grouped`).
2. Go to `wp-admin/admin.php?page=wc-settings&tab=advanced&section=keys` and add a new key with `Read/Write` permissions.
3. Make the following request using the consumer key and secret generated in the previous step, adding the `include_types` param:
```bash
curl --insecure -u [KEY]:[SECRET] --request GET --url 'https://[STORE_URL]/wp-json/wc/v3/products?include_types=simple
```
4. Ensure the result returns only simple products.
5. Repeat the request with multiple types (comma-separated) and check the response makes sense.
6. Try with a non-existing type it should return a 400 response.
7. Try other combinations you can think of and check the response makes sense.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Add the `include_types` param to the `/products` endpoint.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
